core/src/main/scala/org/apache/spark/eventhubs/client/EventHubsClient.scala

```diff
@@ -43,7 +44,13 @@ private[spark] class EventHubsClient(private val ehConf: EventHubsConf)
 
   private implicit val formats = Serialization.formats(NoTypeHints)
 
-  private var client = ClientConnectionPool.borrowClient(ehConf)
+  private var _client: EventHubClient = _
+  private def client = {
+    if (_client == null) {
```

Comment:
>if (_client == null) { [](start = 4, length = 22)

this is being accessed from multiple threads - this block will need synchronization. otherwise - this could lead to borrowing multiple clients.. but keeping track of only 1 of them.